script.js

```diff
@@ -1,4 +1,249 @@
+    sortArrayOutput = playerOneDiceRollList;
+  } else if (gameMode == 2) {
+    playerTwoDiceRollList.sort((a, b) => b - a);
+    sortArrayOutput = playerTwoDiceRollList;
+  }
+  return sortArrayOutput;
+};
+// Concatenate dice values
+var combineNormalDiceRoll = function (numOfDice) {
+  var sortedNormalDiceRoll = "";
+  sortedNormalDiceRoll = sortNormalDice(numOfDice);
+  var combinedValue = 0;
+  if (gameMode == 1) {
+    combinedValue = playerOneDiceRollList.join("");
+  } else if (gameMode == 2) {
+    combinedValue = playerTwoDiceRollList.join("");
+  }
+  return combinedValue;
+};
+// Convert dice string value into integer
+var generateHighestNum = function (numOfDice) {
+  var normalDiceRoll = combineNormalDiceRoll(numOfDice);
+  playerScore = Number(normalDiceRoll);
+  return playerScore;
+};
+
+// Global variable to track each players score
+var playerOneScore = 0;
+var playerTwoScore = 0;
+var playerScore = 0;
+var reversePlayerScore = 0;
+var playerHighScore = "";
+// Global variable for total scores
+var playerOneTotalScore = 0;
+var playerTwoTotalScore = 0;
+
+// Store player one's auto results in global variable for normal mode
+// Output results for player 1
+var playerOneNormalResults = function (numOfDice) {
+  playerHighScore = generateHighestNum(numOfDice);
+  playerOneScore = playerScore;
+  playerOneTotalScore = playerOneTotalScore + playerOneScore;
+  var playerOneOutputMessage = `Player 1: <br> The AI has chosen ${playerOneScore} as your score. <br> <br>It's Player 2's turn. <br> <br> Please click submit to play!`;
+  return playerOneOutputMessage;
+};
+
+// Store player two's auto results in global variable
+// Output each players result
+// Tell players who won
+var playerTwoNormalResults = function (numOfDice) {
+  var playerTwoOutputMessage = "";
+  playerHighScore = generateHighestNum(numOfDice);
+  playerTwoScore = playerScore;
+  playerTwoTotalScore = playerTwoTotalScore + playerTwoScore;
+  var leader = determineNormalLeader();
+  playerTwoOutputMessage = `Player 2: <br> The AI has chosen ${playerTwoScore} as your score. <br> Player 1's score was ${playerOneScore}.  <br><br> ${leader}   <br><br><br> Please enter the number of dice you want to play for the next round!😊`;
+  // Remove existing data in array for next round
+  playerOneDiceRollList.length = 0;
+  playerTwoDiceRollList.length = 0;
+  return playerTwoOutputMessage;
+};
+
+// Determine leader using overall score
+var determineNormalLeader = function () {
+  var findMaxValue = Math.max(playerOneTotalScore, playerTwoTotalScore);
+  if (
+    findMaxValue == playerOneTotalScore &&
+    findMaxValue != playerTwoTotalScore
+  ) {
+    return `Player 1 is in the lead with ${playerOneTotalScore} points! 🎉 <br> <br>Leaderboard: <br>Player 1: ${playerOneTotalScore} points <br>Player 2: ${playerTwoTotalScore} points`;
+  } else if (
+    findMaxValue == playerOneTotalScore &&
+    findMaxValue == playerTwoTotalScore
+  ) {
```

Comment:
Another way you could do this since you only have 2 players will be to just compare the player scores with one another. So for the if statement in line 106, you could just check if player 1 score is higher than player 2 score, instead of checking if player1 score is the highest and if it is not equal to player 2 score

Reply:
I think this is applicable to your logic further down too where you compare the lowest number

Reply:
Ahh I didn't think about that. Thank you! Will remember this for next time as well.